src/utils/utils.py

```diff
@@ -190,105 +194,74 @@ def load_lines(filename: str) -> Iterable[str]:
-    knowledge_to_ix_dic["missing"] = 0
-
-    return {'sents1': sent1s,
-            'sents2': sent2s,
-            'targs': targs,
-            'idxs': idxs,
-            'lex_sem': lex_sem,
-            'pr_ar_str': pr_ar_str,
-            'logic': logic,
-            'knowledge': knowledge,
+        delimiter='\t'):
+    '''Load a tsv and  indexes the columns from the diagnostic tsv.
+        This is only used for MNLI-diagnostic right now.
+
+    Args:
+        data_file: string
+        max_seq_len: int
+        s1_col: string
+        s2_col: string
+        label_col: string
+        label_fn: function
+        skip_rows: list of ints
+        delimiter: string
+
+    Returns:
+        A dictionary of the necessary indexed fields, the tokenized sent1 and sent2
+        and indices
+        Note: If a field in a particular row in the dataset is empty, we return []
+        for that field for that row, otherwise we return an array of ints (indices)
+        Else, we return an array of indices
+    '''
+    # TODO: Abstract indexing layer from this function so that MNLI-diagnostic calls load_tsv
+    assert len(s1_col) > 0 and len(label_col) > 0, "Make sure you passed in column names for sentence 1 and labels"
+    sent1s, sent2s, targs, idxs, lex_sem, pr_ar_str, logic, knowledge = pd.Series(), pd.Series(), pd.Series(), pd.Series(), pd.Series(), pd.Series(), pd.Series(), pd.Series()
+    try:
+        rows = pd.read_csv(data_file, \
+                            sep=delimiter, \
+                            error_bad_lines=False, \
+                            quoting=csv.QUOTE_NONE,\
+                            encoding='utf-8')
+        rows = rows.fillna('')
+        def targs_to_idx(col_name):
+            # This function builds the index to vocab (and its inverse) mapping
+            values = set(rows[col_name].values)
+            vocab = Vocabulary(counter=None)
+            for value in values:
+                vocab.add_token_to_namespace(value, col_name)
+            idx_to_word = vocab.get_index_to_token_vocabulary(col_name)
+            word_to_idx = vocab.get_token_to_index_vocabulary(col_name)
+            rows[col_name] = rows[col_name].apply(lambda x: [word_to_idx[x]] if x != '' else [])
+            return word_to_idx, idx_to_word, rows[col_name]
+
+        sent1s = rows[s1_col].apply(lambda x: process_sentence(x, max_seq_len))
+        sent2s = rows[s2_col].apply(lambda x: process_sentence(x, max_seq_len))
+        labels = rows[label_col].apply(lambda x: label_fn(x))
+        # Build indices for field attributes
+        lex_sem_to_ix_dic, ix_to_lex_sem_dic, lex_sem = targs_to_idx("Lexical Semantics")
+        pr_ar_str_to_ix_di, ix_to_pr_ar_str_dic, pr_ar_str = targs_to_idx("Predicate-Argument Structure")
+        logic_to_ix_dic, ix_to_logic_dic, logic = targs_to_idx("Logic")
+        knowledge_to_ix_dic, ix_to_knowledge_dic, knowledge = targs_to_idx("Knowledge")
+        idxs = rows.index
+
+    except Exception as e:
```

Comment:
See https://docs.python.org/3.1/howto/doanddont.html#except - this is pretty much guaranteed to mask subtle (or not so subtle) bugs. 

What kinds of issues is this expected to trigger on?

Reply:
I don't think I authored this, but I think instead of a try/except, we can add an assertion that the row is the desired length; I think that's what the except was trying to catch originally? The data should be well-formed, in any case, so I'm not totally sure why this was needed or if it was being triggered.

Reply:
Agree that we shouldn't catch general exception though